### DISCUSSION TITLE
File extension denied for sharing

### DISCUSSION THREAD

**Discussion by Nonoss117** [pglombardo/PasswordPusher#1678] [Q&A]
Hi all,
Thank's for the job on this great password pusher !

I tried to share an Excel file containing macros. It seems that files with ".xlsm" extension are not allowed for sharing.
I searched in the doc, but I didn't find a list of authorized/denied extensions for sharing.

Woul'd it be possible to modify authorized extensions files for sharing in the setting.yml or with ENV ?
For example, enable sharing only files with extension in the list, or the contrary, share all extensions except those in the list ?

**Comment by pglombardo** [comment]
Hi @Nonoss117 - thanks for the nice words!

> I tried to share an Excel file containing macros. It seems that files with ".xlsm" extension are not allowed for sharing.

I'm not sure what you mean by this.  Did you get an error in Password Pusher?

For reference, Password Pusher doesn't check or authorize file extensions - any and all file extensions are permitted.

**Reply by Nonoss117** [reply]
Thank's for your quick answer.

This is an example :
I have "Text.xlsx" (23Ko) file for sharing.
I can share it with no problem

I keep the  same file, but Ichange the extension to "xlsm" and then no sharing allowed. Password pusher page stays like this:

![image](https://github.com/pglombardo/PasswordPusher/assets/79055577/09870a5a-7f17-4f3d-98f9-7ab9e6d3c5f5)

That's why I thought that specific extensions where denied.
Do you have an idea of what is wrong ?

**Reply by pglombardo** [reply]
Yeah that is odd.  I just uploaded a file to pwush.com with an `.xlsm` extension without issue.

A few questions:

1.  Are you using pwpush.com or a self-hosted instance?
2. Are there any errors in the browser javascript console?
3. If self-hosting, the docker container logs may have an error - could you post that here?

**Reply by Nonoss117** [reply]
Oups, of course you need more information:
1. It's a self hosted docker instance (on Unraid)
2. I'm not sure, but I don't see any error in the browser
3. logs:

INFO -- : [b70c9afb-a46b-4451-ac34-7868fb1963f0] method=GET path=/ format=html controller=PasswordsController action=new status=200 allocations=9065 duration=97.60 view=23.91 db=15.10 user_id=18 ip=191.96.180.54 forwarded_for=191.96.180.54, 172.18.0.1

INFO -- : [3dd184dc-a2a5-42e6-89f7-1e69b61b05a3] method=GET path=/fr/f/nouveau format=html controller=FilePushesController action=new status=200 allocations=6114 duration=21.78 view=15.85 db=0.48 user_id=18 ip=191.96.180.54 forwarded_for=191.96.180.54, 172.18.0.1

INFO -- : [ae0e3e4f-9b32-4b2c-9ddd-c505f611096b] method=GET path=/ format=html controller=PasswordsController action=new status=0 allocations=269 duration=0.40 view=0.00 db=0.00 ip=[redacted] forwarded_for=[redacted], 172.18.0.1

INFO -- : [af5a628a-836f-4ce9-93fa-07b9bc136052] method=GET path=/fr/utilisateurs/sign_in format=html controller=Users::SessionsController action=new status=200 allocations=4806 duration=16.23 view=15.27 db=0.00 ip=[redacted] forwarded_for=[redacted], 172.18.0.1

INFO -- : [fbbe83f3-cf6c-4ba1-b52a-545b50d412c5] method=POST path=/rails/active_storage/direct_uploads format=json controller=ActiveStorage::DirectUploadsController action=create status=200 allocations=1357 duration=39.66 view=0.43 db=4.47 user_id=18 ip=191.96.180.54 forwarded_for=191.96.180.54, 172.18.0.1

INFO -- : [0bf0f79d-47b0-42a6-a4e0-3147247bf810] method=PUT path=/rails/active_storage/disk/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdDam9JYTJWNVNTSWhlVGRwTm14aFluTnRlR2d5ZEROc2VqZG1jbll5WnpacE1tMXZPUVk2QmtWVU9oRmpiMjUwWlc1MFgzUjVjR1ZKSWpOaGNIQnNhV05oZEdsdmJpOTJibVF1YlhNdFpYaGpaV3d1YzJobFpYUXViV0ZqY205RmJtRmliR1ZrTGpFeUJqc0dWRG9UWTI5dWRHVnVkRjlzWlc1bmRHaHBBaWdmT2cxamFHVmphM04xYlVraUhXWndUMDFSU1hkMmFGSkJhRlZIVEU4M01tbFZNMmM5UFFZN0JsUTZFWE5sY25acFkyVmZibUZ0WlRvS2JHOWpZV3c9IiwiZXhwIjoiMjAyMy0xMS0zMFQxNDoyMjoyOS43MzNaIiwicHVyIjoiYmxvYl90b2tlbiJ9fQ==--

INFO -- : [2a7deac9-33f6-4003-8863-c12f002f87c4] method=GET path=/ format=html controller=PasswordsController action=new status=0 allocations=269 duration=0.48 view=0.00 db=0.00 ip=[redacted] forwarded_for=[redacted], 172.18.0.1

INFO -- : [d3566083-f0a3-4940-b6b5-bd56d7657d73] method=GET path=/fr/utilisateurs/sign_in format=html controller=Users::SessionsController action=new status=200 allocations=4806 duration=13.72 view=13.05 db=0.00 ip=[redacted] forwarded_for=[redacted], 172.18.0.1

**Reply by pglombardo** [reply]
Ok - what version of the container?  Newer versions show the version on the front page in the footer.  It should look like `© 2023 Peter Giacomo Lombardo, v1.34.3`

**Reply by Nonoss117** [reply]
Password pusher version : 1.34.3
Nothing about the version :
![image](https://github.com/pglombardo/PasswordPusher/assets/79055577/c26213a0-685f-406b-8a68-1d8626668d35)
***********************************************************
registry url: https://hub.docker.com/r/pglombardo/pwpush
with tag latest
![image](https://github.com/pglombardo/PasswordPusher/assets/79055577/13e0120c-ad2a-4e77-91fb-c673db0429a8)
***********************************************************
in the log:
![image](https://github.com/pglombardo/PasswordPusher/assets/79055577/4a22ad96-9df4-4940-9c34-fc8e730fef89)

**Reply by pglombardo** [reply]
This all looks good @Nonoss117 - nothing out of the ordinary.  What do you have `PWP__FILES__STORAGE` set to?  local, amazon, google or microsoft?

**Reply by Nonoss117** [reply]
It is set to local:
```yml
  # Environment variable override:
  # PWP__FILES_STORAGE='local'
  #
  storage: 'local'
```

And the folder is accessible read/write from the container

**Reply by pglombardo** [reply]
Could this be a local antivirus or security software?  It's odd that it happens with only `.xslm` files.  Password Pusher doesn't scan extensions and I can't reproduce the issue.

**Reply by Nonoss117** [reply]
I though the same. I tried with disabling all protection (antivirus, firewall, ....), and with my browser (firefox) in "sans échec" mode, all extensions disabled. I also tried with Microsoft Edge.
Same, nothing change.

I tried ".xsle", "xlsv", ".exe" .... all are ok. Yes it's only ".xlsm" the problem.

Never mind, I think I'm the only one with this "problem". Don't waste your time with that. And thank's for your help.

I'll change the extension before sharing,

**Reply by pglombardo** [reply]
Ok - thanks for reporting in any case.

**Reply by anikwai** [reply]
@pglombardo, I was just wondering if it is possible to share SSL private keys using this Password Pusher feature?

**Reply by pglombardo** [reply]
Hi @anikwai - Password Pusher is agnostic to the content you push.  It just handles text and files.  It doesn't check or validate what the text or files are.  Text is stored encrypted in the DB.  Files are stored in the configured storage system.  Text and files are deleted once the push expires.

So you can push anything you like - private keys, ASCII art, zip, jpg etc...  What this what you were asking?

**Reply by anikwai** [reply]
Yes, thanks @pglombardo
